net/veth.go

```diff
@@ -186,6 +186,15 @@ func AttachContainer(netNSPath, id, ifName, bridgeName string, mtu int, withMult
 		return err
 	}
 
+	// set hairpin mode
+	veth, err := netlink.LinkByName(name)
```

Comment:
This should probably be in the closure passed to `CreateAndAttachVeth`, where we already have the veth, and if you return an error there then cleanup will get called